#a########################################################################
# ⭐⭐⭐  Set parameters carefully for satisfied performance!   ⭐⭐⭐ #
#########################################################################

mass        : 1.0 # kg 
gra         : 9.81 
pose_solver : 0    # 1:From ZhepeiWang, 2:From rotor-drag    
ctrl_freq_max   : 200.0
use_bodyrate_ctrl: false
use_position_ctrl: false
use_onboard_landing: false
use_onboard_takeoff : false
max_manual_vel: 2.0
max_angle: 45  # Attitude angle limit in degree. A negative value means no limit.
max_takeoff_land_tilt_angle : -1
max_takeoff_land_hor_acc : -1
standard_voltage: 25.2 # 6S battery
low_voltage: 20.7 # 6S battery
use_smooth_hover_ctrl : true
brake_to_lock_v_xy_norm : 0.1
brake_max_time_s : 1.0 # need to be tuned

rc_reverse: # *
    roll: false
    pitch: false
    yaw: false
    throttle: false

auto_takeoff_land:
    enable: true
    enable_auto_arm: true
    no_RC: false
    takeoff_height: 1.0 # m
    takeoff_land_speed: 0.2 # m/s

thrust_model: # The model that maps thrust signal u(0~1) to real thrust force F(Unit:N): F=K1*Voltage^K2*(K3*u^2+(1-K3)*u). 
    print_value: false # display the value of “thr_scale_compensate” or “hover_percentage” during thrust model estimating.
    accurate_thrust_model: false  # This can always enabled if don't require accurate control performance :-)
    noisy_imu: false
    # accurate thrust mapping parameters
    K1: 0.7583 # Needs precise calibration!
    K2: 1.6942 # Needs precise calibration!
    K3: 0.6786 # Needs precise calibration! K3 equals THR_MDL_FAC in https://docs.px4.io/master/en/config_mc/pid_tuning_guide_multicopter.html.
    # approximate thrust mapping parameters
    # hover_percentage: 0.5 # Thrust percentage in Stabilize/Arco mode # *

    # linear compensation for initial hover percentage 
    # hover_percentage = hp_in_standard_voltage + ( hp_in_low_voltage - hp_in_standard_voltage ) *  (battery_voltage - standard_voltage) / (low_voltage - standard_voltage)
    hp_in_standard_voltage : 0.5 # should use hp and voltage in pair in flight
    hp_in_low_voltage : 0.62  # nagative number means no voltage compensation

gain: 
    # Cascade PID controller. Recommend to read the code.
    Kp0: 1.5
    Kp1: 1.5
    Kp2: 1.5
    Kv0: 2.5
    Kv1: 2.5
    Kv2: 1.5
    # by Steven Cheng
    Kvi0: 0.00
    Kvi1: 0.00
    Kvi2: 0.1 # avoid wrong thrust mapping causing increse height in landing mode :( 
    Kvd0: 0.0
    Kvd1: 0.0
    Kvd2: 0.0
    Kvilim : 1.0
    # ↓↓↓ Only used in rate control mode.
    KAngR: 20.0
    KAngP: 20.0
    KAngY: 4.0

rotor_drag:  
    x: 0.0  # The reduced acceleration on each axis caused by rotor drag. Unit:(m*s^-2)/(m*s^-1).
    y: 0.0  # Same as above
    z: 0.0  # Same as above
    k_thrust_horz: 0.0 # Set to 0 recommended... --

msg_timeout:
    odom: 0.5
    rc:   0.5
    cmd:  0.5
    imu:  0.5
    bat:  0.5
